packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,274 @@
+
+    /**
+     * Optional: Name of the file
+     */
+    public name?: string;
+    /**
+     * A preview of the file which is a lightweight representation.
+     * In case of images this will be a thumbnail/compressed image in base64 encoding.
+     */
+    public preview: string;
+  }
+
+  /**
+   * Checks whether or not visualMedia has user permission
+   *
+   * @beta
+   * @returns Promise that will resolve with true if the user had granted the app permission to media information, or with false otherwise,
+   * In case of an error, promise will reject with the error. Function can also throw a NOT_SUPPORTED_ON_PLATFORM error
+   */
+  export function hasPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+
+    return new Promise<boolean>((resolve) => {
+      resolve(sendAndHandleSdkError('permissions.has', permissions));
+    });
+  }
+
+  /**
+   * Requests user permission for visualMedia
+   * @beta
+   * @returns Promise that will resolve with true if the user consented permission for media, or with false otherwise,
+   * In case of an error, promise will reject with the error. Function can also throw a NOT_SUPPORTED_ON_PLATFORM error
+   */
+  export function requestPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+
+    return new Promise<boolean>((resolve) => {
+      resolve(sendAndHandleSdkError('permissions.request', permissions));
+    });
+  }
+
+  /**
+   * Checks if visualMedia capability is supported by the host
+   * @returns boolean to represent whether media is supported
+   * @beta
+   * @throws Error if {@linkcode app.initialize} has not successfully completed
+   */
+  export function isSupported(): boolean {
+    return ensureInitialized(runtime) && runtime.supports.visualMedia && runtime.supports.permissions ? true : false;
+  }
+
+  /**
+   * To enable this image capability will let the app developer ask the user to get images from camera/local storage
+   * @beta
+   */
+  export namespace image {
+    /**
+     * Capture one or multiple image(s) throughing camera.
+     * @beta
+     * @param imageInputs - The input params to customize the image(s) to be captured
+     * @returns Promise that will resolve with {@link VisualMediaFile[]} object or reject with an error.
+     */
+    export function captureImages(imageInputs: ImageProperties): Promise<VisualMediaFile[]> {
+      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      if (!imageInputs || imageInputs == null || imageInputs.visualMediaCount > 10) {
```

Comment:
>imageInputs.visualMediaCount > 10 [](http://example.com/codeflow?start=49&length=33)

Would you want to provide the developer with a more specific error for this condition? Also would you want to test for <= 0?

Reply:
good point! Updated with the specific error and also added the <= 0 condition.